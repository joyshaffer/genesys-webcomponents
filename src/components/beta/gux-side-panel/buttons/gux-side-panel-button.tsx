import { Component, h, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-side-panel-button.less',
  tag: 'gux-side-panel-button'
})
export class GuxSidePanelButton {
  @Prop()
  icon: string;

  @Prop()
  altText: string;

  @Prop()
  isSelected: boolean = false;

  get buttonClass(): string {
    return this.isSelected ? 'selected' : '';
  }

  render() {
    return (
      <button aria-label={this.altText} class={this.buttonClass}>
        <gux-icon decorative icon-name={this.icon}></gux-icon>
      </button>
    );
  }
}
