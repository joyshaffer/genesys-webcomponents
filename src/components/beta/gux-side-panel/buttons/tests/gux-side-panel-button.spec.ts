import { newSpecPage } from '@stencil/core/testing';
import { GuxSidePanelButton } from '../gux-side-panel-button';

const components = [GuxSidePanelButton];
const language = 'en';

describe('gux-side-panel-button', () => {
  it('should build', async () => {
    const html = `<gux-side-panel-button></gux-side-panel-button>`;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxSidePanelButton);
  });

  describe('#render', () => {
    [{ isSelected: false }, { isSelected: true }].forEach(
      ({ isSelected }, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `<gux-side-panel-button is-selected="${isSelected}"></gux-side-panel-button>`;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      }
    );
  });
});
