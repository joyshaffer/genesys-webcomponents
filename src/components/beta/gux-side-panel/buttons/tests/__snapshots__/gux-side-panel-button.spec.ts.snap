// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-side-panel-button #render should render component as expected (1) 1`] = `
<gux-side-panel-button is-selected="false">
  <button>
    <gux-icon decorative=""></gux-icon>
  </button>
</gux-side-panel-button>
`;

exports[`gux-side-panel-button #render should render component as expected (2) 1`] = `
<gux-side-panel-button is-selected="true">
  <button class="selected">
    <gux-icon decorative=""></gux-icon>
  </button>
</gux-side-panel-button>
`;
